server/src/internalClusterTest/java/org/opensearch/action/admin/indices/datastream/DataStreamTestCase.java

```diff
@@ -30,13 +30,23 @@
 import static org.opensearch.test.OpenSearchIntegTestCase.Scope;
 import static org.hamcrest.Matchers.is;
 
-@ClusterScope(scope = Scope.TEST, numDataNodes = 2)
+@ClusterScope(scope = Scope.TEST, numDataNodes = DataStreamTestCase.DATA_NODE_COUNT)
 public class DataStreamTestCase extends OpenSearchIntegTestCase {
 
+    protected static final int DATA_NODE_COUNT = 2;
+
+    private void resetCluster() {
+        internalCluster().stopAllNodes();
+        internalCluster().startNodes(DATA_NODE_COUNT);
+    }
+
     public AcknowledgedResponse createDataStream(String name) throws Exception {
         CreateDataStreamAction.Request request = new CreateDataStreamAction.Request(name);
         AcknowledgedResponse response = client().admin().indices().createDataStream(request).get();
         assertThat(response.isAcknowledged(), is(true));
+        if (performRemoteStateRestore()) {
```

Comment:
could we have extended this class to do these modifications instead of updating it in place? as we are just add mutations at the end of the methods

Reply:
This is the utility class, actual DataStream's test is in separate class which is extended from this class. If we extend from this class for this change, we will need to change at test classes as well to extend. 

Now after incorporating other comments, now it is single line of change which will be no op for non-remote case and perform reset with remote cluster state case.